pkg/apis/core/v1beta1/defaults_shoot_test.go

```diff
@@ -631,66 +631,51 @@ var _ = Describe("Defaults", func() {
 		})
 	})
 
-	Describe("#SetDefaults_Maintenance", func() {
-		var obj *Maintenance
-
-		BeforeEach(func() {
-			obj = &Maintenance{AutoUpdate: &MaintenanceAutoUpdate{}}
-		})
-
+	Describe("Maintenance defaulting", func() {
 		It("should correctly default the maintenance", func() {
-			obj.TimeWindow = nil
+			obj.Spec.Maintenance = &Maintenance{}
 
-			SetDefaults_Maintenance(obj)
+			SetObjectDefaults_Shoot(obj)
 
-			Expect(obj.TimeWindow).NotTo(BeNil())
-			Expect(obj.TimeWindow.Begin).To(HaveSuffix("0000+0000"))
-			Expect(obj.TimeWindow.End).To(HaveSuffix("0000+0000"))
+			Expect(obj.Spec.Maintenance.TimeWindow).NotTo(BeNil())
+			Expect(obj.Spec.Maintenance.TimeWindow.Begin).To(HaveSuffix("0000+0000"))
+			Expect(obj.Spec.Maintenance.TimeWindow.End).To(HaveSuffix("0000+0000"))
 		})
 	})
 
-	Describe("#SetDefaults_Worker", func() {
-		var obj *Worker
-
-		BeforeEach(func() {
-			obj = &Worker{}
-		})
-
+	Describe("Worker defaulting", func() {
 		It("should set the allowSystemComponents field", func() {
-			obj.SystemComponents = nil
-
-			SetDefaults_Worker(obj)
-
-			Expect(obj.SystemComponents.Allow).To(BeTrue())
+			SetObjectDefaults_Shoot(obj)
+			for i := range obj.Spec.Provider.Workers {
+				a := &obj.Spec.Provider.Workers[i]
+				Expect(a.SystemComponents.Allow).To(BeTrue())
```

Comment:
Let's avoid using a var named `a`:

```suggestion
				worker := &obj.Spec.Provider.Workers[i]
				Expect(a.SystemComponents.Allow).To(BeTrue())
```

```suggestion
				Expect(obj.Spec.Provider.Workers[i].SystemComponents.Allow).To(BeTrue())
```

This comment applied to all other occurrences of variables `a`.